components/pd_client/src/util.rs

```diff
@@ -291,33 +310,32 @@ where
     Req: Clone + Send + 'static,
     F: FnMut(&Client, Req) -> PdFuture<Resp> + Send + 'static,
 {
-    async fn reconnect_if_needed(&mut self) -> bool {
+    async fn reconnect_if_needed(&mut self) -> Result<()> {
         debug!("reconnecting ..."; "remain" => self.reconnect_count);
-
         if self.request_sent < MAX_REQUEST_COUNT {
-            return true;
+            return Ok(());
+        }
+        if self.reconnect_count == 0 {
```

Comment:
I think `reconnect_count` naming is misleading, can we rename it as `remain_reconnect_count` ?